flexget/components/ftp/sftp_client.py

```diff
@@ -26,38 +26,230 @@
 logger = logger.bind(name='sftp_client')
 
 
-class SftpClient:
+class SftpError(Exception):
```

Comment:
Sorry these changes are so huge. There are two reasons:

1) I made an `SftpClientBuilder` class and moved a bunch of logic out of `SftpClient` both for separation of concerns, and to allow for better dependency injection.
2) Objects declared below a given class can't be used as a type annotation in the constructor, which forced me to move SftpClient to the bottom.